app/models/miq_worker/runner.rb

```diff
@@ -83,21 +83,6 @@ def set_connection_pool_size
   # Worker Monitor Methods
   ###############################
 
-  def self.wait_for_worker_monitor?
-    !!@wait_for_worker_monitor
-  end
-
-  class << self
-    attr_writer :wait_for_worker_monitor
-  end
-
-  def my_monitor_started?
-    return @monitor_started unless @monitor_started.nil?
-    return false if     server.nil?
-    return false unless server.reload.started?
-    @monitor_started = true
-  end
```

Comment:
This is method that unravels the thread.  This method waits for the server to start, whereas now we wait for the just the worker to start in `do_delay_startup_for_vim_broker` here: https://github.com/ManageIQ/manageiq/pull/15654/files#diff-a7ce0fdab9c89206c1d9565fd7eddc31R324

Additionally, waiting for the server to start has the strange side effect of handling a worker that says it "needs to wait for the broker", but where a broker will never actually start because it's not needed.  This happens because MiqVimBrokerWorker.workers will be 0, thus the server will never wait for it, and then will start nornmally.  This is handled directly now in the gate to  call `do_delay_startup_for_vim_broker` here: https://github.com/ManageIQ/manageiq/pull/15654/files#diff-a7ce0fdab9c89206c1d9565fd7eddc31R149